File: CPP/Program/PetShop.cpp
#include <iostream>
#include <string>
#include <list>

using namespace std;

class PetShop{
    private:
        int id;
        string nama;
        string kategori;
        int harga;
    
    public:
    PetShop(){
    }

    PetShop(int i, string n, string k, int h){
        id = i;
        nama = n;
        kategori = k;
        harga = h;
    }

    void setId(int i){
        id = i;
    }

    void setNama(string n){
        nama = n;
    }

    void setKategori(string k){
        kategori = k;
    }

    void setHarga(int h){
        harga = h;
    }

    int getId(){
        return id;
    }

    string getNama(){
        return nama;
    }

    string getKategori(){
        return kategori;
    }

    int getHarga(){
        return harga;
    }

    void tampilkanData(){
        cout << "ID         : " << id << endl;
        cout << "Nama       : " << nama << endl;
        cout << "Kategori   : " << kategori << endl;
        cout << "Harga      : " << harga << endl;
        cout << "--------------------------" << endl;
    }

    static void tampilkanSemuaData(list<PetShop>& data){
        for(auto& item : data){
            item.tampilkanData();
        }
    }

    static void tambahData(list<PetShop>& data, PetShop p){
        data.push_back(p);
    }

    static void ubahData(list<PetShop>& data, int id, string nama, string kategori, int harga){
        auto it = data.begin();
        int flag = 0;
        while(it != data.end() && flag == 0){
            if(it->getId() == id){
                it->setNama(nama);
                it->setKategori(kategori);
                it->setHarga(harga);
                flag = 1;
            }
            ++it;
        }
        if(flag == 0){
            cout << "Data tidak ditemukan" << endl;
        }
    }

    static void hapusData(list<PetShop>& data, int id){
        auto it = data.begin();
        int flag = 0;
        while(it != data.end() && flag == 0){
            if(it->getId() == id){
                data.erase(it);
                flag = 1;
            }
            ++it;
        }
        if(flag == 0){
            cout << "Data tidak ditemukan" << endl;
        }
    }

    static void cariData(list<PetShop>& data, int id){
        auto it = data.begin();
        int flag = 0;
        while(it != data.end() && flag == 0){
            if(it->getId() == id){
                it->tampilkanData();
                flag = 1;
            }
            ++it;
        }
        if(flag == 0){
            cout << "Data tidak ditemukan" << endl;
        }
    }

    ~PetShop(){
    }
};

File: PHP/Program/PetShop.php
<?php
class PetShop {
    // atribut yang digunakan
    private $id;
    private $nama;
    private $kategori;
    private $harga;
    private $gambar;

    // constructor
    public function __construct($id = null, $nama = null, $kategori = null, $harga = null, $gambar = null) {
        $this->id = $id;
        $this->nama = $nama;
        $this->kategori = $kategori;
        $this->harga = $harga;
        $this->gambar = $gambar;
    }

    // setter dan getter

    // setter id
    public function setId($id) {
        $this->id = $id;
    }

    // setter nama produk
    public function setNama($nama) {
        $this->nama = $nama;
    }

    // setter kategori
    public function setKategori($kategori) {
        $this->kategori = $kategori;
    }

    // setter harga
    public function setHarga($harga) {
        $this->harga = $harga;
    }

    // setter gambar
    public function setGambar($gambar) {
        $this->gambar = $gambar;
    }

    // getter id
    public function getId() {
        return $this->id;
    }

    // getter nama produk
    public function getNama() {
        return $this->nama;
    }

    // getter kategori
    public function getKategori() {
        return $this->kategori;
    }

    // getter harga
    public function getHarga() {
        return $this->harga;
    }

    // getter gambar
    public function getGambar() {
        return $this->gambar;
    }

    // method untuk upload gambar
    public function uploadGambar($file) {
        $target_dir = "images/";
        
        // buat direktori untuk menyimpan file image
        if (!is_dir($target_dir)) {
            mkdir($target_dir, 0777, true);
        }
        
        // buat nama file gambar
        $unique_name = $target_dir . time() . '_' . basename($file["name"]);
        if (move_uploaded_file($file["tmp_name"], $unique_name)) {
            $this->setGambar($unique_name);
        }
            
    }
}
?>